scripts/speech_recognition/convert_to_tarred_audio_dataset.py

```diff
@@ -644,17 +568,28 @@ def _read_manifest(self, manifest_path: str, config: ASRTarredDatasetConfig):
 
         return entries, total_duration, filtered_entries, filtered_duration
 
-    def _write_to_tar(self, tar, audio_filepath: str, squashed_filename: str) -> None:
-        if (codec := self.config.force_codec) is None or audio_filepath.endswith(f".{codec}"):
+    def _write_to_tar(
+        self, tar, audio_filepath: str, squashed_filename: str, duration: float = None, offset: float = 0
+    ) -> None:
+        if ((codec := self.config.force_codec) is None or audio_filepath.endswith(f".{codec}")) and not duration:
             # Add existing file without transcoding.
             tar.add(audio_filepath, arcname=squashed_filename)
         else:
-            # Transcode to the desired format in-memory and add the result to the tar file.
+            # Read audio file
             audio, sampling_rate = soundfile.read(audio_filepath, dtype=np.float32)
+            # Calculate start and end points for slicing
+            start_sample = int(offset * sampling_rate)
+            end_sample = int((offset + duration) * sampling_rate) if duration else None
+            audio = audio[start_sample:end_sample]
+            # Transcode and write to tar
             encoded_audio = BytesIO()
-            if codec == "opus":
-                kwargs = {"format": "ogg", "subtype": "opus"}
+            if codec is not None:
```

Comment:
Actually we could do a fast-path here: `if codec=None and offset=0 and duration=None`, instead of reading+writing with soundfile (triggering encoding and decoding) just read the raw audio file bytes and write them to tar file directly. Will likely speed up data tarring by a non-trivial factor. Up to you.